deploy-tool/cert/cert.go

```diff
@@ -14,12 +14,17 @@ import (
 	"github.com/caddyserver/certmagic"
 )
 
-// IssueCertificate asks the Let's Encrypt ACMEApiEndpoint to validate and sign a certificate for a site using the DNS01 strategy
-func IssueCertificate(deployConfig config.DeployConfig, site string, privKey *ecdsa.PrivateKey) (string, error) {
-	dnsProvider, err := dns.GetACMEDNSProvider(deployConfig)
+// CertificateManager is a structure that defines the Certificate Issuance and renewal implementation
+type CertificateManager struct {
+	Config config.DeployConfig
+}
```

Comment:
Per jsha's advice in Slack, I think there should be a `New` function to instantiate a `CertificateManager` that takes a `config.DeployConfig`. Also I don't think anything outside of the `CertificateManager` implementation ever needs to see its inner config object, so it could be private (i.e., rename to `config`).